core/domain/story_services.py

```diff
@@ -174,6 +175,14 @@ def apply_change_list(story_id, change_list):
                 elif (change.property_name ==
                       story_domain.STORY_PROPERTY_THUMBNAIL_FILENAME):
                     story.update_thumbnail_filename(change.new_value)
+                    thumbnail_url = (
+                        'http://localhost:8181/assetsdevhandler/story/%s'
+                        '/assets/thumbnail/%s'
+                        % (story_id, change.new_value))
```

Comment:
This is not right. You can check `fs_services` and `fs_domain` to find out how the URL is generated in general (for both dev and prod). 
`fs_domain` consists of the function that generates the URL and saves images for the different classes and `fs_services` sets the class based on Dev or Prod envt.

Reply:
updated